src/Algebra/Module/Structures.agda

```diff
@@ -0,0 +1,262 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some algebraic structures defined over some other structure
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary using (Rel; Setoid; IsEquivalence)
+
+module Algebra.Module.Structures
+  {m ℓm} {M : Set m} (_≈ᴹ_ : Rel M ℓm)
+  where
+
+open import Algebra.Bundles
+open import Algebra.Core
+import Algebra.Definitions as Defs
+import Algebra.Module.Definitions.Left as L
+import Algebra.Module.Definitions.Right as R
+open import Algebra.Structures
+open import Data.Product using (_,_; proj₁; proj₂)
+open import Level using (Level; _⊔_)
+
+private
+  variable
+    r ℓr s ℓs : Level
+
+module _ (semiring : Semiring r ℓr) where
```

Comment:
So the problem with this organisation of parameters is that the order goes: `_≈ᴹ_ semiring +ᴹ ...` which is a little bit weird... I know it loses the module parameter but I think it'd be much cleaner to have `semiring _≈ᴹ_ +ᴹ ...`?